### PR TITLE
Minor enhancement to storage instructions + various nits

### PR DESCRIPTION
**Pull request by mohammadfawaz**
* Defined the `STATE[..]` syntax.
* Also some minor nits all over the spec.

### CODE REVIEW

**Review by adlerjohn — APPROVED** [review]

File: src/protocol/id/utxo.md [hunk 1]
```diff
@@ -9,10 +9,10 @@ Is represented as an _outpoint_: a pair of [transaction ID](./transaction.md) as
...
 
 1. the sender address as `byte[32]`,
-1. the recipient address as `byte[32]`,
-1. the [Message nonce](#message-nonce) as `byte[32]`,
-1. the amount being sent with the message as `uint64`,
-1. the message data as `byte[]`
+2. the recipient address as `byte[32]`,
+3. the [Message nonce](#message-nonce) as `byte[32]`,
+4. the amount being sent with the message as `uint64`,
+5. the message data as `byte[]`
```
Comment (xgreenx) [inline src/protocol/id/utxo.md:15]: Better to have `1.`. Markdown will increment it automatically

File: src/protocol/tx_format/transaction.md [hunk 2]
```diff
@@ -31,15 +31,15 @@ Transaction is invalid if:
 When serializing a transaction, fields are serialized as follows (with inner structs serialized recursively):
 
 1. `uint8`, `uint16`, `uint32`, `uint64`: big-endian right-aligned to 8 bytes.
-1. `byte[32]`: as-is.
-1. `byte[]`: as-is, with padding zeroes aligned to 8 bytes.
+2. `byte[32]`: as-is.
+3. `byte[]`: as-is, with padding zeroes aligned to 8 bytes.
```
Comment (xgreenx) [inline src/protocol/tx_format/transaction.md:35]: The same here

File: src/vm/index.md [hunk 3]
```diff
@@ -78,17 +78,17 @@ Every time the VM runs, a single monolithic memory of size `VM_MAX_RAM` bytes is
 To initialize the VM, the following is pushed on the stack sequentially:
 
 1. Transaction hash (`byte[32]`, word-aligned), computed as defined [here](../protocol/id/transaction.md).
-1. [`MAX_INPUTS`](../protocol/tx_format/constants.md) pairs of `(asset_id: byte[32], balance: uint64)`, of:
+2. [`MAX_INPUTS`](../protocol/tx_format/constants.md) pairs of `(asset_id: byte[32], balance: uint64)`, of:
```
Comment (xgreenx) [inline src/vm/index.md:81]: The same in this file

File: src/vm/instruction_set.md [hunk 4]
```diff
@@ -1107,10 +1108,10 @@ For output with contract ID `MEM[$rA, 32]`, increase balance of asset ID `MEM[$r
 A [call frame](./index.md#call-frames) is pushed at `$sp`. In addition to filling in the values of the call frame, the following registers are set:
 
 1. `$fp = $sp` (on top of the previous call frame is the beginning of this call frame)
-1. Set `$ssp` and `$sp` to the start of the writable stack area of the call frame.
-1. Set `$pc` and `$is` to the starting address of the code.
-1. `$bal = $rB` (forward coins)
-1. `$cgas = $rD` or all available gas (forward gas)
+2. Set `$ssp` and `$sp` to the start of the writable stack area of the call frame.
+3. Set `$pc` and `$is` to the starting address of the code.
```
Comment (xgreenx) [inline src/vm/instruction_set.md:1112]: The same in this file

**Review by xgreenx — COMMENTED** [review]
I see that you did it by the tool, which automatically increases the list's numeral. I think better to use `1.` everywhere, but it is up to you.

File: src/protocol/id/utxo.md (hunk 1 above)
Comment (mohammadfawaz) [reply src/protocol/id/utxo.md:15]: I'll revert this change. I see that this is better for diffs when a new item gets added in the middle of the list.

**Review by xgreenx — APPROVED** [review]

**Pull request merged**